tests/installation/ipxe_install.pm

```diff
@@ -187,7 +205,7 @@ sub run {
     if (get_var('VIRT_AUTOTEST')) {
         #it is static menu and choose the TW entry to start installation
         enter_o3_ipxe_boot_entry if get_var('IPXE_STATIC');
-        assert_screen([qw(load-linux-kernel load-initrd)], 240);
+        check_screen([qw(load-linux-kernel load-initrd)], 240);
```

Comment:
Why use ```check_screen``` instead ?

Reply:
The two needles can't be always caught because sol output flushes fast at this time , but it won't impact the following test which will check further screens.

Reply:
It seems our ipmi backend test runs always use assert_screen which looks good. I do not think you need to make this change.

Reply:
Sorry, this does not convince me. I do not think it is always good, just like what I met ever on vh015. This is  a minor improvement and won't hurt stability but increase pass ratio and robustness.

Reply:
I created some needles for some special machines, so we got 95%(I guess) pass rate with assert_screen(). Why I used 'assert_screen' instead of 'check_screen' is that I would like to confirm which medium the host really installs, including the location, the build number and any error when loading kernel files.

The advantage of "check_screen" is 5% more pass rate and no creating needles for vh015 alone if you don't botter the status of loading kernel files, just in the way hana team or core team install their hosts.

Reply:
In summary, "assert_screen" is more suitable for use in OSD because I have created needles which will make all SUTs work; while "check_screen" is more suitable for one's personal openqa server because lack of needles.

Reply:
1. I think assert_screen is better for test run.
2. I do not see a need to replace it with check_screen from experience on OSD.
3. I think assert_screen is more suitable for most cases.

Reply:
> In summary, "assert_screen" is more suitable for use in OSD because I have created needles which will make all SUTs work; while "check_screen" is more suitable for one's personal openqa server because lack of needles.

@Julie-CAO Thanks for the reply. 
Let me add more info here for better understanding the situation. For vh015, I do not think it is fully because of missing needles, because it passes at 80%+ with this needle check , while for about 20%-, it fails at this needle check but pass at next needle check. So I think the main reason is that the screen sometimes flush fast or the sol output is problematic, and thus needle match is failing. It is hard to ensure 100% pass here by adding needles. And these BJ machines impact our daily work efficiency, so I think it is worthwhile to figure out some workaround for them.

Besides, it is easy to change it back. I can do it if there are good reasons. But I do not quite understand the benefit here and your conclusion that having this is better for OSD test. IMHO, it is almost the same.
Main reason: compared to assert_screen, check_screen is also doing the check until it timeouts, so it is not ignoring this check. When it does not pass this check, it will go to next screen check which is the next step for installation, so if this step indeed fails, but not by missing some needles for specific machines, eg kernel load problem, the test will still fail at closest next assert_screen. And for reviewer, it will be same easy to tell the problem as before, because for such case, it will fail at this assert_screen until its final flush of screen, just the same as how it fail for the next screen check for this PR. And the benefit is to get more chance for passing.

One more thing, also minor -- when we get more OSD machines which can be a continuous process, do you prefer adding potential new needles for this?

Reply:
> Main reason: compared to assert_screen, check_screen is also doing the check until it timeouts, so it is not ignoring this check.

Yes, I agree. "check_screen" takes similiar effect with "assert_screen" with enough needles.

I also thought about changing "check_screen" to "assert_screen" when a SUT on OSD failed to match this needle in a build, but finally it passed, so I did not make this change at last.

With autoyast installation, we need at least one "assert_screen" to make sure that it is a real installation rather booting from existing OS directly, so if consequent 

```
assert_screen([qw(network-config-created loading-installation-system sshd-server-started autoyast-installation)], 300);
```
can resolve this issue, I think there is no difference between "check_screen" or "assert_screen" when loading kernel files.

Reply:
> One more thing, also minor -- when we get more OSD machines which can be a continuous process, do you prefer adding potential new needles for this?

I did add extra needles when I added unreal2/3 and when SLEM 5.5 was supported. It is a little tricky but I have been skilled :rofl:  And even with replacing with "check_screen", I think these needles would still be helpful, because we will know what the failure is to fail to start an autoyast installation.

Reply:
I checked `network-config-created` and think it matches fresh installation only. `loading-installation-system sshd-server-started autoyast-installation` should be same too if I have a good memory.

since 

```
assert_screen([qw(network-config-created loading-installation-system sshd-server-started autoyast-installation)], 300);
```
 can differenciate a fresh installation from booting from existing OS, I prefer to using "check_screen" at here. As of whether adding more needles, it depends on if the user need it. it is not mandotory.

Reply:
Thanks, then we keep this way.